ocaml/xapi/message_forwarding.ml

```diff
@@ -688,6 +688,7 @@ functor
       include Local.Task
 
       let cancel ~__context ~task =
+        Context.with_tracing ~__context __FUNCTION__ @@ fun __context ->
```

Comment:
Is it necessary to instrument `cancel` in message_forwarding? This function is here to decide where `cancel` would be executed. And I see you are already instrucmenting `cancel` in `xapi_task.ml` already. Does this mean that we will see two traces in jaeger? Is that useful to have?

Reply:
They are actually the same trace. When we are making the `rpc` we retrieve the `traceparent` from the context and we pass it on. I forgot to save an image of how the trace looks, so until I make another one, this is in short the result:

---> Xapi_task.cancel
---> ---> Message_forwarding.Task.cancel
---> ---> ---> <Client>.Task.cancel                     --- where <Client> is where the `cancel` is executed.

Reply:
It is not a problem if we nest tracing, the more information we have, the better. We are not required to instrument all (nested) functions though.